implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpConfig.java

```diff
@@ -229,15 +231,16 @@ default HistogramFlavor histogramFlavor() {
 
     /**
      * Configures the histogram flavor to use on a per-meter level. This will override the
-     * {@link #histogramFlavor()} configuration for matching Meters. The key is used to do
-     * an exact match on the Meter's name.
-     * @return mapping of meter name to histogram flavor
+     * {@link #histogramFlavor()} configuration for matching Meters. By default, the key
+     * is used to do an exact match on the Meter's name.
+     * @return mapping of meter ID to histogram flavor
      * @since 1.15.0
      * @see #histogramFlavor()
      */
-    default Map<String, HistogramFlavor> histogramFlavorPerMeter() {
-        return getStringMap(this, "histogramFlavorPerMeter", HistogramFlavor::fromString)
-            .orElse(Collections.emptyMap());
+    default Function<Meter.Id, HistogramFlavor> histogramFlavorPerMeter() {
```

Comment:
I made the signature `Function<Meter.Id, HistogramFlavor>` for the most flexibility, but perhaps `<String, HistogramFlavor>` is better where the `String` is the meter name. While programmatically the implementation of the `Function` could do anything, we usually try to keep it reasonable to be configured from a Properties/Map source. I don't see a reasonable implementation from properties that can be deserialized into a Function accepting a Meter.Id. I also wonder if users in practice would want to make this configuration on some basis other than name anyway.